technique/sans/instrument.py

```diff
@@ -685,41 +839,43 @@ def convert_file(file_path):
                 if "trans" in row:
                     del row["trans"]
 
-                out.write('    {}("{}", '.format(header, title))
+                out.write(f'    {header}(title="{title}", ')
 
                 if "pos" in row:
-                    out.write('"{}", '.format(row["pos"]))
+                    out.write(f'position="{row["pos"]}", ')
                     del row["pos"]
 
-                row = {k: row[k] for k in row if row[k].strip() != ""}
-                for k in row.keys():
-                    if row[k].upper() == "TRUE":
-                        row[k] = True
-                    elif row[k].upper() == "FALSE":
-                        row[k] = False
+                row = {key: row[key] for key in row if row[key].strip() != ""}
+                for key in row.keys():
+                    if row[key].upper() == "TRUE":
+                        row[key] = True
+                    elif row[key].upper() == "FALSE":
+                        row[key] = False
                     else:
                         try:
-                            row[k] = ast.literal_eval(row[k])
+                            row[key] = ast.literal_eval(row[key])
                         except ValueError:
-                            row[k] = "\"" + row[k] + "\""
+                            row[key] = "\"" + row[key] + "\""
                             continue
-                params = ", ".join([k + "=" + str(v)
-                                    for (k, v) in sorted(row.items())])
-                out.write('{})\n'.format(params))
+                params = ", ".join([key + "=" + str(value)
+                                    for (key, value) in sorted(row.items())])
+                out.write(f'{params})\n')
 
     @staticmethod
-    def printsamplepars():
+    def print_sample_pars():
         """Display the basic sample parameters on the console."""
         pars = gen.get_sample_pars()
         for par in ["Geometry", "Width", "Height", "Thick"]:
-            info("{}={}".format(par, pars[par.upper()]))
+            info(f"{par}={pars[par.upper()]}")
 
     def enumerate_dae(self):
-        """List the supported DAE modes on this beamline."""
+        """List the supported DAE modes on this beamline.
+        Warning: Some methods may not be implemented from the base class
+        """
         return [x[10:] for x in dir(self) if x.startswith("setup_dae_")]
```

Comment:
We can remove `10` as a magic number here

```suggestion
        setup_dae = "setup_dae_"
        return [x[len(setup_dae):] for x in dir(self) if x.startswith(setup_dae)]
```